name.abuchen.portfolio/src/name/abuchen/portfolio/util/TextUtil.java

```diff
@@ -362,4 +363,82 @@ public static String escapeHtml(String input)
                         .replace("\"", "&quot;") //$NON-NLS-1$ //$NON-NLS-2$
                         .replace("'", "&#39;"); //$NON-NLS-1$ //$NON-NLS-2$ Ï
     }
+
+    /**
+     * Determines if a search text matches a numeric value. Supports partial
+     * matches (prefixes), locale-specific formatting, and separators
+     * (i.e. 2,000 = 2000)
+     */
+    public static boolean isNumericSearchMatch(String searchText, double value)
+    {
+        if (searchText == null || searchText.isEmpty())
+            return false;
+
+        // Get locale-specific decimal and grouping separators
+        char decimalSeparator = new DecimalFormatSymbols(Locale.getDefault()).getDecimalSeparator();
+        char groupingSeparator = new DecimalFormatSymbols(Locale.getDefault()).getGroupingSeparator();
+
+        // Remove grouping separators from search text
+        String cleanedSearchText = searchText.replace(String.valueOf(groupingSeparator), "");
+
+        // Check if the search text is a valid number format
+        if (!isValidNumberFormat(cleanedSearchText, decimalSeparator))
+            return false;
+
+        // Format the value to a string using the current locale (using absolute
+        // value)
+        DecimalFormat df = new DecimalFormat("#.###");
+        df.setGroupingUsed(false); // Disable grouping
+        String formattedValue = df.format(Math.abs(value));
+
+        // Remove any negative sign from search text for comparison
+        if (cleanedSearchText.startsWith("-"))
+            cleanedSearchText = cleanedSearchText.substring(1);
+
+        // Split both strings into whole and decimal parts
+        boolean searchHasDecimal = cleanedSearchText.contains(String.valueOf(decimalSeparator));
+        boolean valueHasDecimal = formattedValue.contains(String.valueOf(decimalSeparator));
+
+        String[] searchParts = searchHasDecimal
+                        ? cleanedSearchText.split(Pattern.quote(String.valueOf(decimalSeparator)), 2)
+                        : new String[] { cleanedSearchText };
+        String[] valueParts = valueHasDecimal ? formattedValue.split(Pattern.quote(String.valueOf(decimalSeparator)), 2)
+                        : new String[] { formattedValue };
```

Comment:
You will noticed in the merged code, that I replaced the regular expression with an indexOf. Then we do not have to have the "contains" and the regular expressions (which is compiled on every search).